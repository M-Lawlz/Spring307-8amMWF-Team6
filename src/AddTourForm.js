import React from 'react';
import App from "firebase/app";

class AddTourForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tourId: "",
      userEmail: "",
      location: "",
      uploadDate: "",
      videoUrl: "",
      description: "",
      username:"",
      comments:[],
      currentTourValue: 0,
      user : null,
      userData : null
    }
  }

  async componentDidMount() {
    await App.auth().onAuthStateChanged((user) => {
      if (user) {
        this.setState({ user: user });
        this.fetchUserInfo();
      }
    });
    const firebase = require("firebase");
    const db = firebase.firestore();
    const toursDb = db.collection("Tours");

    toursDb.get().then(snapshot => {
        snapshot.forEach(doc => {
            if(doc.data().tourId > this.state.currentTourValue) {
                this.setState({
                    currentTourValue : doc.data().tourId
                });
            }
        });
    }).catch((error) => {
      console.error("Error: " + error.message);
    });
  }

  fetchUserInfo = () => {
    const userDoc = App.firestore()
      .collection("users")
      .doc(this.state.user.email);
    userDoc
      .get()
      .then((doc) => {
        if (doc.exists) {
          this.setState({userData: doc.data()});
        }
      })
      .catch((error) => {
        console.error("Error: ", error.message);
      });
  };

  updateInput = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  addTour = e => {
    e.preventDefault();
    const firebase = require("firebase");
    const db = firebase.firestore();
    var date = new Date();
    var timeRn =
            ("00" + (date.getMonth() + 1)).slice(-2) + "/" +
            ("00" + date.getDate()).slice(-2) + "/" +
            date.getFullYear();
    db.collection("Tours").doc((this.state.currentTourValue + 1).toString()).set({
      location: this.state.location,
      description: this.state.description,
      videoUrl: this.state.videoUrl,
      uploadDate: timeRn,
      userEmail: this.state.userEmail,
      uploaderUsername: this.state.userData.username,
      comments: [],
      tourId: this.state.currentTourValue + 1,
      tourLikes: 0
    }).then((info) => {
      alert("Tour successfully uploaded!");
      /* allows for the tour to be shown immediately in the search bar */
      window.location.reload(true);
    }).catch(function(error) {
      console.log("Error with uploading to database! "  + error);
    });

    this.setState({
      tourId: "",
      userEmail: "",
      location: "",
      uploadDate: "",
      videoUrl: "",
      description: "",
      comments:[],
      currentTourValue : this.state.currentTourValue + 1
    });
  };

  render(){
    return (
      
      <section>
        <form onSubmit={this.addTour}>
        
        <h3>Tour Name</h3>
        <label></label>
        <input type="text" 
        name="name" 
        placeholder="Name of Tour" 
        required
        onChange={this.updateInput}
        />
         <h3>Location</h3>
        <label></label>
        <input type="text" 
        name="location" 
        placeholder="Tour Location"
        required
        onChange={this.updateInput}
        value={this.state.location}/>
        <h3>Description</h3>
        <label></label>
        <input type="text" 
        name="description"
        placeholder="Tour Description" 
        required
        onChange={this.updateInput} 
        value={this.state.description}/>
         <h3>Embed Url</h3>
        <label></label>
        <input type="text" 
        name="videoUrl" 
        placeholder="Embed Link"
        required
        onChange={this.updateInput} 
        value={this.state.videoUrl}/>
        <h3>Email</h3> 
        <label></label>
        <input type="email" 
        name="userEmail" 
        placeholder="email"
        required
        onChange={this.updateInput} 
        value={this.state.userEmail}/>
        
        <button type="submit" disabled={!this.state.userData}>Add Tour</button>  
        </form>
      </section>

      )
  }}

export default AddTourForm
