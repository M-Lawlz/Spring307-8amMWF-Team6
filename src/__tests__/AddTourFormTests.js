import AddTourForm from "../AddTourForm";
import React from "react";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";


test("AddTourForm renders successfully.", () => {
  const addTourForm = render(<AddTourForm />);
  expect(toJson(addTourForm)).toMatchSnapshot();
});

const wrapper = shallow(<AddTourForm />);
describe("updateInput", () => {
  it("should call setState", () => {
    const expected = {
      description: "test",
      location: "test",
      uploadDate: "test"
    
    };
    
    setTimeout(() => {
    wrapper.instance().updateInput({target: { value: 'test', name: 'test'}});
    expect(wrapper.state.test).toEqual('test');
  }, 0);
  });
});
