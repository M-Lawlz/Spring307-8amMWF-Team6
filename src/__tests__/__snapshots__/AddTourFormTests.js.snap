// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTourForm renders successfully 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Tour Name
  </label>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name of Tour"
    required={true}
    type="text"
  />
  <label>
    Location
  </label>
  <input
    name="location"
    onChange={[Function]}
    placeholder="Tour Location"
    required={true}
    type="text"
    value=""
  />
  <label>
    Description
  </label>
  <input
    name="description"
    onChange={[Function]}
    placeholder="Tour Description"
    required={true}
    type="text"
    value=""
  />
  <label>
    Embed Url
  </label>
  <input
    name="videoUrl"
    onChange={[Function]}
    placeholder="Embed Link"
    required={true}
    type="text"
    value=""
  />
  <label>
    Upload Date
  </label>
  <input
    name="uploadDate"
    onChange={[Function]}
    placeholder="Current Date"
    required={true}
    type="date"
    value=""
  />
  <button
    type="submit"
  >
    Add Tour
  </button>
</form>
`;
